### PR TITLE
Add blockpull/blockcommit cases

### PR DESCRIPTION
**Pull request by chloerh**
- when guest image with "file" disk has "block" backing file
- when guest image with "file" disk has "gluster" backing file
- when guest image with "block" disk has "block" backing file

Signed-off-by: haizhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
Test result:
---
 (1/6) type_specific.local.backingchain.blockcommand.blockpull.base_top.positive_test.file_disk.block_bk: PASS (49.15 s)
 (2/6) type_specific.local.backingchain.blockcommand.blockpull.base_top.positive_test.file_disk.gluster_bk: PASS (40.28 s)
 (3/6) type_specific.local.backingchain.blockcommand.blockpull.base_top.positive_test.block_disk.block_bk: PASS (50.81 s)
 (4/6) type_specific.local.backingchain.blockcommand.blockcommit.base_top.positive_test.file_disk.block_bk: PASS (52.57 s)
 (5/6) type_specific.local.backingchain.blockcommand.blockcommit.base_top.positive_test.file_disk.gluster_bk: PASS (43.91 s)
 (6/6) type_specific.local.backingchain.blockcommand.blockcommit.base_top.positive_test.block_disk.block_bk: PASS (49.94 s)

File: libvirt/tests/cfg/backingchain/blockcommand.cfg [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    variants:
+        - positive_test:
+
+    variants blockcommand:
+        - blockpull:
+            variants check_func:
+                - base_top:
+                    base = 0
+#                - middle_top:
+#                    base = 2
```
Comment (Yingshun) [inline libvirt/tests/cfg/backingchain/blockcommand.cfg]: need to be removed?

File: libvirt/tests/cfg/backingchain/blockcommand.cfg [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        - blockcommit:
+            variants check_func:
+                - base_top:
+                    top = 0
+                    base = 0
+#                - middle_middle:
+#                    top = 1
+#                    base = 3
```
Comment (Yingshun) [inline libvirt/tests/cfg/backingchain/blockcommand.cfg]: Same as above.

**Comment by chloerh** [comment]
@lento-sun Would you like to review this pr

File: libvirt/tests/src/backingchain/blockcommand.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+            disk_target = '/dev/%s/%s' % (vg_name, vol_bk)
+            src_vol = '/dev/%s/%s' % (vg_name, vol_disk)
+
+            # Command of creating block disk
+            cmd_create_img = 'qemu-img create -f qcow2 -b %s %s' % (disk_target, src_vol)
+
+        # Setup gluster
+        elif disk_src == 'gluster':
+            host_ip = libvirt.setup_or_cleanup_gluster(
```
Comment (Yingshun) [inline libvirt/tests/src/backingchain/blockcommand.py]: setup_or_cleanup_gluster is moved to gluster.py.

File: libvirt/tests/cfg/backingchain/blockcommand.cfg (hunk 1 above)
Comment (chloerh) [reply libvirt/tests/cfg/backingchain/blockcommand.cfg]: These comments are examples for future editors to refer to.

File: libvirt/tests/src/backingchain/blockcommand.py (hunk 3 above)
Comment (chloerh) [reply libvirt/tests/src/backingchain/blockcommand.py]: Thanks for the remider.

File: libvirt/tests/src/backingchain/blockcommand.py [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+                **virsh_dargs
+            )
+
+            # Get path of each snapshot file
+            snaps = virsh.domblklist(vm_name, debug=True).stdout.splitlines()
+            for line in snaps:
+                if line.startswith(('hd', 'sd', 'vd')):
+                    file_to_del.append(line.split()[-1])
+
+        qemu_img_cmd = 'qemu-img info -U --backing-chain %s' % snapshot_image_list[-1]
```
[the remark this answers is not in the document]
Comment (chloerh) [reply libvirt/tests/src/backingchain/blockcommand.py]: Updated

File: libvirt/tests/src/backingchain/blockcommand.py [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+
+    logging.debug(pattern)
+    exist = re.search(pattern, img_info, re.DOTALL)
+    logging.debug(exist)
+    return exist
+
+
+def check_bc_base_top(command, vmxml, dev, bc_chain):
+    """
+    Check backing chain info after blockpull/commit from base to top
```
[the remark this answers is not in the document]
Comment (chloerh) [reply libvirt/tests/src/backingchain/blockcommand.py]: Comments edited. It seems more work to split this function into 2.

File: libvirt/tests/cfg/backingchain/blockcommand.cfg [hunk 6]
```diff
@@ -0,0 +1,41 @@
+- backingchain.blockcommand:
+    type = blockcommand
+    start_vm = 'no'
+    status_error = 'no'
+    opts = '--verbose --wait '
+    variants:
+        - file_disk:
+            disk_type = 'file'
+            variants:
```
[the remark this answers is not in the document]
Comment (chloerh) [reply libvirt/tests/cfg/backingchain/blockcommand.cfg]: I guess. But we'll need a formal procedure. Not in this pr.

**Comment by lento-sun** [comment]
I am ok with this, and for ceph part, will open a trello card to follow.

**Pull request merged**